### DISCUSSION TITLE
Android Update Issue

### DISCUSSION THREAD

**Discussion by supern64** [ppy/osu#21895] [Q&A] [answered]
I don't know if this really qualifies, but today I tried to upgrade from 2022.1208.0 to 2022.1228.0 and my phone apparently wouldn't install the update. Looking further into it, it seems like the signatures for the older one do not match the newer one. Did something about the signing process change? I know I could reinstall it and get it working again but I just want to know.

**Comment by bdach** [comment, marked as answer]
The deployment did happen using a slightly different procedure than usual so yes something may have changed wrt signing. @peppy was the same signing key used?

**Reply by peppy** [reply]
should have been. in theory.

can we get more details from the user on why it wouldn't install? a screenshot of the detailed error or anything?

**Reply by bdach** [reply]
@supern64 do you still have the error message on hand?

**Reply by supern64** [reply]
I seem to be able to replicate it... maybe I could try getting logs from ADB.

**Reply by supern64** [reply]
```
Performing Streamed Install
adb: failed to install sh.ppy.osulazer-1.apk: Failure [INSTALL_FAILED_UPDATE_INCOMPATIBLE: Existing package sh.ppy.osulazer signatures do not match newer version; ignoring!]
```
this is the only line it spits out after i run `adb install -r [apk file]`

**Reply by bdach** [reply]
Hm. I did see an installation error when installing the APK during testing but presumed it was because I was making releases locally with the same version but different signature... Another epic fail on my behalf I guess.

**Reply by Susko3** [reply]
The APK was signed with the default auto-generated key. Looks like the keystore wasn't properly referenced?
More info on keystores in: .NET https://learn.microsoft.com/en-us/dotnet/maui/android/deployment/publish-cli?view=net-maui-7.0

![image](https://user-images.githubusercontent.com/16479013/210058234-d9b9f140-ef86-45fd-b08a-3a5306c68a7b.png)
https://www.virustotal.com/gui/file/3e3d0c8072c669721819db7d81ea1db98e3fbdedc242344ad8745f807afbb3d9/details

Previous versions would use
![image](https://user-images.githubusercontent.com/16479013/210058799-6c398a7a-ab84-4fb7-ad79-62fd7c39bc42.png)
https://www.virustotal.com/gui/file/d7603003409b6259861dddb11d594d3e5860bf133f36d3daa9d1874455835827/details

**Reply by peppy** [reply]
That gives a lot of confidence since it produced no errors... Will revisit today and see if I can figure it out.

**Reply by peppy** [reply]
This should now be fixed, please download the apk again.
